extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php

```diff
@@ -6,7 +6,44 @@ public function appreciate() {
 		global $wgUser;
 
 		$this->request->assertValidWriteRequest( $wgUser );
-		//TODO: do something with apprecation
+
+		$revisionId = $this->request->getInt( 'revision' );
+		$revision = Revision::newFromId( $revisionId );
+		if ( $revision instanceof Revision ) {
```

Comment:
`newFromId` can return null or Revision, it should be enough to check for falsy value